fission-cli/library/Fission/CLI/IPFS/Executable.hs

```diff
@@ -59,17 +66,57 @@ place' ::
   -> m ()
 place' host = do
   logUser $ "🪐 Downloading managed IPFS for " <> textDisplay host
+  unpack =<< download (IPFS.Version 0 9 0) host
+  configure
 
-  IPFS.BinPath    ipfsPath <- Path.globalIPFSBin
-  File.Serialized lazyFile <- ensureM . runBootstrapT . ipfsCat $ IPFS.binCidFor host
-
-  logDebug $ "Writing IPFS binary to " <> Text.pack ipfsPath
-  ipfsPath `forceWrite` Lazy.toStrict lazyFile
-
-  void . Turtle.chmod Turtle.executable $ Turtle.decodeString ipfsPath
-
+download ::
+  ( MonadGitHub      m
+  , MonadLogger      m
+  , MonadRaise       m
+  , m `Raises` ClientError
+  )
+  => IPFS.Version
+  -> OS.Supported
+  -> m Lazy.ByteString
+download version os = do
+  logDebug $ "⬇️  Downloading go-ipfs " <> display version <> " for " <> display os
+  ensureM . GitHub.sendRequest $ IPFS.getRelease IPFS.Release {..}
+
+unpack :: (MonadIO m, MonadLogger m, MonadEnvironment m) => Lazy.ByteString -> m ()
+unpack tarGz = do
+  IPFS.BinPath ipfsPath <- Path.globalIPFSBin
+  tmp                   <- Path.globalTmpDir
+
+  let
+    tmpTar      = tmp </> "go-ipfs-release.tar.gz"
+    source      = Turtle.decodeString (tmp </> "go-ipfs" </> "ipfs")
+    destination = Turtle.decodeString ipfsPath
+
+  -- FileSystem
+  logDebug @Text "💗 Decompressing archive..."
+  File.forceWrite tmpTar (Lazy.toStrict $ GZip.decompress tarGz)
+
+  logDebug @Text "🦖 Untarring..."
+  liftIO $ Tar.extract tmp tmpTar
+  void $ Turtle.chmod Turtle.executable source
+
+  logDebug @Text "🚎 Moving IPFS into place..."
+  createDirectoryIfMissing True $ dropFileName ipfsPath
+  Turtle.mv source destination
+
+  logDebug @Text "🧹 Cleaning up..."
+  Turtle.rmtree $ Turtle.decodeString tmp
+
+configure ::
+  ( MonadEnvironment m
+  , MonadLocalIPFS   m
+  , MonadLogger      m
+  , MonadRescue      m
+  , m `Raises` IPFS.Error
+  )
+  => m ()
+configure = do
   logUser @Text "🎛️  Configuring managed IPFS"
-
   void IPFS.Config.init
   void IPFS.Config.enableRelay
 
```

Comment:
I broke the stages out here, mainly because there's more manual work now with the tarball